isovar/genetic_code.py

```diff
@@ -0,0 +1,165 @@
+
+class GeneticCode(object):
+    def __init__(self, name, start_codons, stop_codons, codon_table):
+        self.name = name
+        self.start_codons = set(start_codons)
+        self.stop_codons = set(stop_codons)
+        self.codon_table = dict(codon_table)
+        self._check_codons()
+
+    def _check_codons(self):
+        """
+        If codon table is missing stop codons, then add them.
+        """
+        for stop_codon in self.stop_codons:
+            if stop_codon in self.codon_table:
+                if self.codon_table[stop_codon] != "*":
+                    raise ValueError(
+                        "Stop codon '%s' must translate to '*'" % (
+                            stop_codon,))
+            else:
+                self.codon_table[stop_codon] = "*"
+        for start_codon in self.start_codons:
+            if start_codon not in self.codon_table:
+                raise ValueError(
+                    "Start codon '%s' missing from codon table" % (
+                        start_codon,))
+        for codon, amino_acid in self.codon_table.items():
+            if amino_acid == "*" and codon not in self.stop_codons:
+                raise ValueError(
+                    "Non-stop codon '%s' can't translate to '*'" % (
```

Comment:
Could something like "Codon '%s' not found in stop_codons, but codon_table indicates that it should be" be a little clearer?